src/main/java/org/plumelib/util/CombinationIterator.java

```diff
@@ -32,11 +33,14 @@
  */
 public class CombinationIterator<T> implements Iterator<List<T>> {
 
+  /** The original, indexable list of candidate collections (one per position). */
+  private final List<Collection<T>> listOfCollectionsOfCanditates;
```

Comment:
Why is a new variable needed?  A type-checker should not require the addition of new, unnecessary data structures.  How can you make the type-checker pass without algorithmic changes?